src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.cs

```diff
@@ -243,6 +243,30 @@ public override void Initialize(Solution solution)
             {
                 var connection = pooledConnection.Connection;
 
+                // Enable exclusive locking to prevent expensive file open/close/locking/unlocking syscalls,
+                // since in this scenario the databases are not shared between different processes.
+                // Different connections within the same process are compatible with exclusive mode, because of
+                // shared cache mode is used to open connections (see SqlConnection.Create).
+                // See https://www.sqlite.org/sharedcache.html#shared_cache_locking_model for information on interactions
+                // between shared cache and locking modes.
+                // Exclusive locking is efficient for any journal mode, but brings an additional improvement to write-ahead 
+                // logging, removing the requirement of using shared memory for wal-index, thus allowing SHM-incapable 
+                // filesystems to be used (see https://sqlite.org/wal.html#use_of_wal_without_shared_memory).
+                connection.ExecuteCommand("pragma locking_mode=exclusive", throwOnError: false);
+
+                // Enable write-ahead logging to increase write performance by reducing amount of disk writes,
+                // by combining writes at checkpoint, salong with using sequential-only writes to populate the log.
+                // Also, WAL allows for relaxed ("normal", see below) "synchronous" mode without compromising durability.
```

Comment:
I think durability is compromised.  But that's ok, we don't need durability for the persistence service.  It's ok if the writes don't actually make it all the way to the DB.  The higher level layers will recover fine if htat happens.